Rx/v2/src/rxcpp/operators/rx-sequence_equal.hpp

```diff
@@ -170,44 +173,47 @@ struct sequence_equal : public operator_base<bool>
 };
 
-template<class OtherObservable, class Coordination>
+template<class OtherObservable, class BinaryPredicate, class Coordination>
 class sequence_equal_factory
 {
     typedef rxu::decay_t<OtherObservable> other_source_type;
     typedef rxu::decay_t<Coordination> coordination_type;
+    typedef rxu::decay_t<BinaryPredicate> predicate_type;
     
     other_source_type other_source;
     coordination_type coordination;
+    predicate_type pred;
     
 public:
-    sequence_equal_factory(other_source_type t, coordination_type sf)
+    sequence_equal_factory(other_source_type t, predicate_type p, coordination_type sf)
         : other_source(std::move(t))
         , coordination(std::move(sf))
+        , pred(std::move(p))
     {
     }
     
     template<class Observable>
     auto operator()(Observable&& source)
-        ->      observable<bool, sequence_equal<rxu::value_type_t<rxu::decay_t<Observable>>, Observable, other_source_type, Coordination>> {
-        return  observable<bool, sequence_equal<rxu::value_type_t<rxu::decay_t<Observable>>, Observable, other_source_type, Coordination>>(
-                                 sequence_equal<rxu::value_type_t<rxu::decay_t<Observable>>, Observable, other_source_type, Coordination>(std::forward<Observable>(source), other_source, coordination));
+        ->      observable<bool, sequence_equal<rxu::value_type_t<rxu::decay_t<Observable>>, Observable, other_source_type, BinaryPredicate, Coordination>> {
+        return  observable<bool, sequence_equal<rxu::value_type_t<rxu::decay_t<Observable>>, Observable, other_source_type, BinaryPredicate, Coordination>>(
+                                 sequence_equal<rxu::value_type_t<rxu::decay_t<Observable>>, Observable, other_source_type, BinaryPredicate, Coordination>(std::forward<Observable>(source), other_source, pred, coordination));
     }
 };
 
 }
 
 template<class OtherObservable>
 inline auto sequence_equal(OtherObservable&& t)
-    ->      detail::sequence_equal_factory<OtherObservable, identity_one_worker> {
-    return  detail::sequence_equal_factory<OtherObservable, identity_one_worker>(std::forward<OtherObservable>(t), identity_current_thread());
+    ->      detail::sequence_equal_factory<OtherObservable, rxu::equal_to<>, identity_one_worker> {
+    return  detail::sequence_equal_factory<OtherObservable, rxu::equal_to<>, identity_one_worker>(std::forward<OtherObservable>(t), rxu::equal_to<>(), identity_current_thread());
+}
+
+template<class OtherObservable, class BinaryPredicate>
+inline auto sequence_equal(OtherObservable&& t, BinaryPredicate&& pred)
+    ->      detail::sequence_equal_factory<OtherObservable, BinaryPredicate, identity_one_worker> {
+    return  detail::sequence_equal_factory<OtherObservable, BinaryPredicate, identity_one_worker>(std::forward<OtherObservable>(t), std::forward<BinaryPredicate>(pred));
 }
 
-template<class OtherObservable, class Coordination>
-inline auto sequence_equal(OtherObservable&& t, Coordination&& sf)
```

Comment:
if the coordination overloads are missing, then the operator cannot be used to coordinate the values from source that are producing values from different threads.